src/librustdoc/html/static/main.js

```diff
@@ -2952,6 +2937,35 @@ function defocusSearchBar() {
 
     onHashChange(null);
     window.onhashchange = onHashChange;
+
+    function loadScript(url) {
+        const script = document.createElement('script');
+        script.src = url;
+        document.head.append(script);
+    }
+
+    window.searchLoaded = false;
+    function loadSearch() {
+        if (!window.searchLoaded) {
+            window.searchLoaded = true;
+            loadScript(window.searchJS);
+        }
+    }
+
+    // Assumes `crates{version}.js` has already been loaded, setting the global const `crates`.
+    addSearchOptions(crates);
```

Comment:
I don't like much to "assume" things are there in an async world. Instead, please call the function to init the crates lists from the `crates{version}.js` file directly.

Reply:
Ok nevermind. It's fine as is. However, please use `window.crates` or equivalent. I don't like the usage of global with such common names, it makes the readibility a lot more problematic.